source/backend/api/Services/DocumentFileService.cs

```diff
@@ -89,333 +91,509 @@ public IList<T> GetFileDocuments<T>(FileType fileType, long fileId)
+            // Step 1 - Save the pimsDocument
+            PimsDocument pimsDocument = CreatePimsDocument(uploadRequest);
+            _documentQueueRepository.SaveChanges();
 
-            DocumentUploadRelationshipResponse relationshipResponse = new()
+            // Step 2 - Create the File Document
+            PimsDispositionFileDocument newFileDocument = new()
             {
-                UploadResponse = uploadResult,
+                DispositionFileId = dispositionFileId,
+                DocumentId = pimsDocument.DocumentId,
             };
+            _dispositionFileDocumentRepository.AddDispositionDocument(newFileDocument);
+            _documentQueueRepository.SaveChanges();
 
-            // Throw an error if Mayan returns a null document. This means it wasn't able to store it.
-            ValidateDocumentUploadResponse(uploadResult);
-
-            if (uploadResult.Document is not null && uploadResult.Document.Id != 0)
+            // Step 3 - Queue the Document for processing
+            PimsDocumentQueue queueDocument = new()
             {
-                PimsDispositionFileDocument newDocument = new()
-                {
-                    DispositionFileId = dispositionFileId,
-                    DocumentId = uploadResult.Document.Id,
-                };
-                newDocument = _dispositionFileDocumentRepository.AddDispositionDocument(newDocument);
-                _dispositionFileDocumentRepository.CommitTransaction();
-
-                relationshipResponse.DocumentRelationship = mapper.Map<DocumentRelationshipModel>(newDocument);
-            }
+                DocumentId = pimsDocument.DocumentId,
+                DispositionFileDocumentId = newFileDocument.DispositionFileDocumentId,
+                Document = await uploadRequest.File.GetBytes(),
+                FileName = uploadRequest.File.FileName,
+                DocumentMetadata = JsonSerializer.Serialize(uploadRequest.DocumentMetadata),
+            };
 
-            return relationshipResponse;
+            _documentQueueRepository.Add(queueDocument);
+            _documentQueueRepository.SaveChanges();
+
+            // All good here.
+            transaction.Commit();
+
+            return;
         }
 
         public async Task<ExternalResponse<string>> DeleteResearchDocumentAsync(PimsResearchFileDocument researchFileDocument)
         {
             Logger.LogInformation("Deleting PIMS document for single research file");
             User.ThrowIfNotAllAuthorized(Permissions.DocumentDelete, Permissions.ResearchFileEdit);
 
-            var relationshipCount = _documentRepository.DocumentRelationshipCount(researchFileDocument.DocumentId);
-            if (relationshipCount == 1)
-            {
-                return await documentService.DeleteDocumentAsync(researchFileDocument.Document);
-            }
-            else
+            var result = new ExternalResponse<string>() { Status = ExternalResponseStatus.NotExecuted };
```

Comment:
hmm, if a status of NotExecuted is returned in the case of a document with no mayanId, does that cause problems on the frontend?

Reply:
no issues.